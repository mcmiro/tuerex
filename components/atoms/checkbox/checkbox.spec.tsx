import * as React from 'react';
import { screen, render, fireEvent } from '@testing-library/react';
import Checkbox from './';

describe('Test Checkbox component', () => {
  it('Should pass when name added', () => {
    render(<Checkbox name="test" />);

    const checkbox = screen.getByRole('checkbox');
    expect(checkbox).toBeDefined();
    expect(checkbox).toMatchSnapshot();
  });

  it('Should pass when component is not checked', () => {
    const changeMock = jest.fn();
    render(
      <Checkbox name="test" onCheck={changeMock} data-testid="my-checkbox" />
    );

    const checkbox = screen.getByRole('checkbox') as HTMLInputElement;
    expect(checkbox.checked).toBe(false);
  });

  it('Should pass when component click checkbox and change checked to true', async () => {
    const changeMock = jest.fn();
    render(
      <Checkbox name="test" onCheck={changeMock} data-testid="my-checkbox" />
    );

    const checkbox = screen.getByRole('checkbox') as HTMLInputElement;
    fireEvent.click(checkbox);
    expect(changeMock).toHaveBeenCalledTimes(1);
    expect(checkbox).toBeChecked();
  });
});
