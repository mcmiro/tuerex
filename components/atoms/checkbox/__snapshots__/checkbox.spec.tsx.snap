// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Checkbox component Should pass when name added 1`] = `
<input
  class="form-checkbox w-6 h-6 ring-1 ring-black text-black bg-primary-100 disabled:ring-gray-500 disabled:text-gray-500"
  id=":r0:"
  name="test"
  type="checkbox"
/>
`;
